montepy/data_inputs/fill.py

```diff
@@ -104,39 +126,29 @@ def _parse_cell_input(self, key, value):
         value : str
             the value given in the cell
         """
-
-        def get_universe(value):
-            if ":" in value["data"].nodes:
-                self._parse_matrix(value)
-            else:
-                data = value["data"]
-                try:
-                    val = data[0]
-                    val._convert_to_int()
-                    assert val.value >= 0
-                    self._old_number = val
-                except (TypeError, AssertionError) as e:
-                    raise ValueError(
-                        f"The fill universe must be a valid integer ≥ 0, {data} was given"
-                    )
-                # ensure only one universe is given
-                if (
-                    len(data) >= 2
-                    and isinstance(data[1], syntax_node.ValueNode)
-                    and "(" != data[1].value
-                ):
-                    raise ValueError(
-                        f"Fill cannot have two universes in this format. {data.format()} given"
-                    )
-
         data = value["data"]
-        if not isinstance(data, syntax_node.ListNode):
-            data = data.nodes
-        if "(" in data:
-            get_universe(value)
-            trans_data = value["data"][
-                list(value["data"]).index("(") + 1 : list(value["data"]).index(")") - 1
-            ]
+        if not isinstance(data, syntax_node.SyntaxNode):
+            data = self._precondition_tree(data)
+        if len(data["indices"]) > 0:
+            self._parse_matrix(value)
+        else:
+            uni_data = data["universes"]
+            try:
+                val = uni_data[0]
+                val._convert_to_int()
```

Comment:
if we need to access this from outside the objects, shouldn't the method just be public?